### ISSUE TITLE
'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected version spec in futures; python_version == "2.7" at ; python_version == "2.7"

### ISSUE THREAD

**Issue by SimonPe** [timothycrosley/isort#652]
offending code:
```
      install_requires=['futures; python_version == "2.7"'],
```
This piece of code (line 58 in setup.py) is not commited in the git repo but is released in v4.3.0 on pypi (which isn't tagged in this git repo eighther).

This breaks our CI pipelines.

**Comment by Azd325** [comment]
Maybe it has to defined like this way
```cfg
    extras_require={
        ':python_version=="2.7"': [
            'futures',
        ],
    }
```

**Comment by dwighthubbard** [comment]
The install_requires line looks correct.  However the published package would need to be generated by setuptools 36.4 or higher or the wheel package won't work correctly.  

Currently the ensurepip module that python 3.6.4 ships with is bundled with setuptools 28.8.0 so setuptools would need to be updated in any new virtualenvs created with 3.6.4 before the package is generated.  In order for the wheel package to work correctly with the environment markers in the install_requires.

**Comment by zachriggle** [comment]
Can confirm this is affecting Pwndbg

**Comment by zerothi** [comment]
Confirm with dependency from pylint, `pip install pylint` will also fail on 2.7 due to isort dependency.

**Comment by timothycrosley** [comment]
This should now be fixed in release 4.3.1, Thank you!
